microsoft/botframework-solutions
# [Enterprise Bot][TypeScript] Add test with mocks for escalateDialog and mainDialog

## Description
- Add `testBase.js` helpers to encapsulate the use of mocked services.
- Add tests for `escalateDialog.ts`.
- Add tests for `mainDialog.ts`.
- Add mocks for new tests.

## Related Issue




#704

## Testing Steps


1. Open a terminal in `templates/Enterprise-Template/src/typescript/enterprise-bot/`
1. Run `npm i` and `npm run build` to install dependencies and build the project.
1. Run `npm test`.
1. You should see four tests passing, one for Escalate dialog and three for Main dialog.

![image](https://user-images.githubusercontent.com/12394167/52509188-f4681100-2bd5-11e9-9334-c6085ff24323.png)

## Checklist

If this contains changes that needs to be replicated between the Enterprise Template <-> Virtual Assistant

If you have updated responses or `.lu` files:
